src/hidimstat/desparsified_lasso.py

```diff
@@ -4,141 +4,111 @@
-        clf = Lasso(alpha=alpha, precompute=gram_, max_iter=max_iter, tol=tol)
-
-    else:
-
-        ValueError("The only regression method available is 'lasso'")
-
-    clf.fit(X_new, y)
-    z = y - clf.predict(X_new)
-
-    omega_diag_i = n_samples * np.sum(z**2) / np.dot(y, z) ** 2
-
-    return z, omega_diag_i
+from hidimstat.noise_std import reid
+from hidimstat.stat_tools import pval_from_two_sided_pval_and_sign
+from hidimstat.stat_tools import pval_from_cb
 
 
 def desparsified_lasso(
     X,
     y,
     dof_ajdustement=False,
-    confidence=0.95,
     max_iter=5000,
     tol=1e-3,
-    residual_method="lasso",
     alpha_max_fraction=0.01,
+    eps=1e-2,
+    tol_reid=1e-4,
+    n_split=5,
     n_jobs=1,
-    memory=None,
+    seed=0,
     verbose=0,
+    group=False,
+    cov=None,
+    noise_method="AR",
+    order=1,
+    fit_Y=True,
+    stationary=True,
 ):
-    """Desparsified Lasso with confidence intervals
+    """
+    Desparsified Lasso with confidence intervals
+
+    Algorithm based on Algorithm 1 of d-Lasso and d-MTLasso in
+    :cite:`chevalier2020statistical`.
 
     Parameters
     ----------
     X : ndarray, shape (n_samples, n_features)
-        Data.
+        Input data matrix.
 
-    y : ndarray, shape (n_samples,)
-        Target.
+    y : ndarray, shape (n_samples,) or (n_samples, n_times)
+        Target vector for single response or matrix for multiple
+        responses.
 
     dof_ajdustement : bool, optional (default=False)
-        If True, makes the degrees of freedom adjustement (cf. [4]_ and [5]_).
-        Otherwise, the original Desparsified Lasso estimator is computed
-        (cf. [1]_ and [2]_ and [3]_).
-
-    confidence : float, optional (default=0.95)
-        Confidence level used to compute the confidence intervals.
-        Each value should be in the range [0, 1].
+        If True, applies degrees of freedom adjustment.
+        If False, computes original Desparsified Lasso estimator.
 
     max_iter : int, optional (default=5000)
-        The maximum number of iterations when regressing, by Lasso,
-        each column of the design matrix against the others.
+        Maximum iterations for Nodewise Lasso regressions.
 
     tol : float, optional (default=1e-3)
-        The tolerance for the optimization of the Lasso problems: if the
-        updates are smaller than `tol`, the optimization code checks the
-        dual gap for optimality and continues until it is smaller than `tol`.
-
-    residual_method : str, optional (default='lasso')
-        Method used for computing the residuals of the Nodewise Lasso.
-        Currently the only method available is 'lasso'.
+        Convergence tolerance for optimization.
 
     alpha_max_fraction : float, optional (default=0.01)
-        Only used if method='lasso'.
-        Then alpha = alpha_max_fraction * alpha_max.
+        Fraction of max lambda used for Lasso regularization.
```

Comment:
Terminology problem: we need to decide whether we call it alpha or lambda. I think that sklearn has gone for alpha, we should follow that.

Reply:
yes, it's alpha. I will replace lambda by alpha.

Reply:
see PR #132 and #125